integrations/nvidia.md

```diff
@@ -0,0 +1,195 @@
+description: Use NVIDIA models with Haystack.
+authors:
+    - name: deepset
+      socials:
+        github: deepset-ai
+        twitter: deepset_ai
+        linkedin: deepset-ai
+pypi: https://pypi.org/project/haystack-ai
+repo: https://github.com/deepset-ai/haystack
+type: Model Provider
+report_issue: https://github.com/deepset-ai/haystack/issues
+logo: /logos/nvidia.png
+version: Haystack 2.0
+toc: true
+---
+### **Table of Contents**
+- [Overview](#overview)
+- [Installation](#installation)
+- [Usage](#usage)
+- [License](#license)
+
+## Overview
+
+ [NVIDIA AI Foundation Models](https://www.nvidia.com/en-us/ai-data-science/foundation-models/) allow you to reach optimal performance on NVIDIA accelerated infrastructure. With pretrained generative AI models, enterprises can create custom models faster and take advantage of the latest training and inference techniques. And with NVIDIA AI Foundation Endpoints, their applications can be connected to these models running on a fully accelerated stack to test performance.
+
+This integration allows you to use NVIDIA Foundation Models in your Haystack piplines.
+
+
+In order to use this integration, you'll need a [NVIDIA API key](https://org.ngc.nvidia.com/setup). Set it as an environment variable, `NVIDIA_API_KEY`.
+
+## Installation
+
+```bash
+pip install nvidia-haystack
+```
+
+## Usage
+### Components
+This integration introduces the following components:
+
+- `NvidiaTextEmbedder`: 
+    A component for embedding strings, using NVIDIA AI Foundation embedding models.
+
+    For models that differentiate between query and document inputs,
+    this component embeds the input string as a query.
+  
+- `NvidiaDocumentEmbedder`:
+    A component for embedding documents, using NVIDIA AI Foundation embedding models.
+
+- `NvidiaGenerator`:     A component for generating text using generative models provided by NVIDIA AI Foundation Endpoints.
+
+- `NvidiaGeneratorModel` and `NvidiaEmbeddingModel` are enums indicating which NVIDIA models are currently supported by the integration.
+
+## Use the components on their own:
+  
+### `NvidiaTextEmbedder`:
+
+```python
+from haystack_integrations.components.embedders.nvidia import NvidiaEmbeddingModel, NvidiaTextEmbedder
+
+
+text_to_embed = "I love pizza!"
+
+text_embedder = NvidiaTextEmbedder(model=NvidiaEmbeddingModel.NVOLVE_40K)
+text_embedder.warm_up()
+
+print(text_embedder.run(text_to_embed))
+# {'embedding': [-0.02264290489256382, -0.03457780182361603, ...}
+
+```
+### `NvidiaDocumentEmbedder`:
+```python
+from haystack.dataclasses import Document
+from haystack_integrations.components.embedders.nvidia import NvidiaEmbeddingModel, NvidiaDocumentEmbedder
+
+documents = [Document(content="Pizza is made with dough and cheese"),
+             Document(content="Cake is made with floud and sugar"),
+             Document(content="Omlette is made with eggs")]
+
+
+
+document_embedder = NvidiaDocumentEmbedder(model=NvidiaEmbeddingModel.NVOLVE_40K)
```

Comment:
Do users usually know which model this `NVOLVE_40K` refers to?

Reply:
it's this one - https://catalog.ngc.nvidia.com/orgs/nvidia/teams/ai-foundation/models/nvolve-40k

Adding a link is probably a good idea just to make that clearer.